bzt/utils.py

```diff
@@ -627,6 +632,30 @@ def install(self):
             except BaseException as exc:
                 raise exc
 
+    def install_with_mirrors(self, dest, suffix):
+        self.log.info("Will try to install %s into %s", self.tool_name, dest)
+        downloader = request.FancyURLopener()
+        tool_dist = tempfile.NamedTemporaryFile(suffix=suffix, delete=False)  # delete=False because of Windows
+        mirrors = self.mirror_manager.mirrors()
+        while True:
```

Comment:
Difference is in code readability and explicitness. "for" loop would clearly say that you're iterating over mirrors. "while" loop says that you're doing something and code reader should search inside the loop for lopp stop conditions.